reconcile/terraform_cloudflare_dns.py

```diff
@@ -87,11 +88,43 @@ def run(self, dry_run: bool) -> None:
 
     @defer
     def run_with_defer(self, dry_run: bool, defer: Callable) -> None:
-        vault_settings = get_app_interface_vault_settings()
-        secret_reader = create_secret_reader(use_vault=vault_settings.vault)
+
+        settings = self._get_app_interface_settings()
+
+        if not settings.settings:
+            raise RuntimeError("App interface setting undefined.")
+
+        if not settings.settings[0].vault:
+            raise RuntimeError("App interface vault setting undefined.")
+
+        if not settings.settings[0].cloudflare_dns_zone_max_records:
+            raise RuntimeError(
+                "Default max number of records for Cloudflare DNS zone undefined."
```

Comment:
Any reason we can't just have some sane default, rather than making the integration error out if someone hasn't set this?

Reply:
Updated

Reply:
Am I missing something, or will the integration still error out if the max records settings isn't set?